### PR TITLE
Mike - Fixed bug where you can add class to a different quarter's schedule

### LINKED ISSUE #6: BUG: Add course dialog allows adding course to the wrong schedule
### Summary

When using the button on the main course search results to add a course to a personal schedule, the button allows the user to add courses to a schedule for a different quarter than the course.  This should not be allowed. Instead, the modal that shows a list of personal schedules to choose from *should only* show personal schedules for the quarter that matches the course being added.  If no such personal schedules exist, the model should say, for example (if the quarter is F24): 

>No personal schedules exist for quarter F24.
> `[Create One]`

Where `[Create One]` is a button that navigates to the create page for personal schedules.

### Expected Behavior

When adding a course from a given quarter to a personal schedule, only schedules that match the quarter of the course should be presented as options.

### Current / Observed Behavior

When adding a course from a given quarter to a personal schedule, all personal schedules are presented as options. This results in incorrect behavior, where one of two things happens:
* Either there is a POST error when the enrollment code doesn't exist in the target quarter
* A course with the target enrollment code (e.g. 23456) is added, but it could be an *entirely different course* from the one intended.

### Steps to Reproduce

1. Login to the https://courses.dokku-00.cs.ucsb.edu website
2. Create at least one personal schedule for a given quarter (e.g. W22 as shown below)
    
    image

4. Search for a course from a *different* quarter (e.g. S24)
    
    image

6. Find a course you can add with the Green Add button.  Click the button to add that course to a personal schedule.

   
8. The modal pops up as shown below.  Select a schedule for the wrong quarter.
   
    image

10. Observe that you get either an error or you get a course other than the one you intended showing up on your schedule.  
    
Here's what the error looks like:
       
image

Here's what a wrong course looks like. This happens when by coincidence, there is a course with the same enrollment
code in the target quarter (but it could be an entirely different course!  In this case, we tried to add ANTH 3 from F24 but ended up with ANTH 5 from W22.)

image

image
  
image

    

# Acceptance Criteria

- [x] When you click the add button for a course in a given quarter, say F24, in the modal that pops up, all of the personal schedules for that quarter are shown, and ONLY the personal schedules for that quarter are shown.   (NOTE: This could be factored out into it's own issue an PR).
- [x] When you click the add button for a course in a given quarter, say F24, if there are no personal schedules for that quarter that already exist, the model says: "There are no personal schedules for F24" (or whatever the quarter is), and there is a button [Create Personal Schedule] that when you click it, navigates to the `https://courses-qa.dokku-00.cs.ucsb.edu/personalschedules/create` page.    (NOTE: This could be factored out into its own issue and PR)
- [ ] Nice to have, but not essential: add an optional parameter to the https://courses-qa.dokku-00.cs.ucsb.edu/personalschedules/create url e.g. `https://courses-qa.dokku-00.cs.ucsb.edu/personalschedules/create?qtr=f24` so that the quarter is preset to the selected one, and use this on the modal.  (This should definitely be factored out into a separate issue and PR.)

### PR DESCRIPTION
**Pull request by MeDotE**
This fixes the bug described in issue #6 where choosing a course from the main course search page would allow you to add that class to a schedule for a different quarter, which would result in either error or the adding of a random(?) class. 

Screenshot 2024-05-28 at 21 39 21

image (1)

Screenshot 2024-05-28 at 21 40 57

Screenshot 2024-05-28 at 21 41 12

Screenshot 2024-05-28 at 21 41 22

As you can see in these images, if I try to add a class from W22, only the W22 schedule will show up. The same is for other quarters. 

Dokku deployment: [https://proj-courses-medote-dev.dokku-03.cs.ucsb.edu](https://proj-courses-medote-dev.dokku-03.cs.ucsb.edu)

WARNING: This PR also contains the changes to `frontend/.storybook/main.js`, `frontend/package-lock.json`, `frontend/package.json`, and `frontend/.storybook/webpack-plugin-fail-build-on-warning.js` because I initially committed changed from when I applied the storybook run fix as provided by Kevin L. in a previous PR and in the team slack. 

Closes #6

### CODE REVIEW

**Review by MeDotE — COMMENTED** [review]
i made a mistake but i already typed all this so imma just leave it as a draft...

**Comment by MeDotE** [comment]
ready to be reviewed now

**Review by a-k-u-m-a-r — DISMISSED** [review]
LGTM! Prompted to create new schedule, if it does not exist for the class. Otherwise, the behavior sent me to a personal schedule that I already created

**Review by pconrad — DISMISSED** [review]
code LGTM... will test before merging

**Comment by pconrad** [comment]
Deployed to: https://courses-qa.dokku-03.cs.ucsb.edu/

File: frontend/package.json [hunk 1]
```diff
@@ -54,7 +55,7 @@
     "test": "react-scripts test",
     "eject": "react-scripts eject",
     "coverage": "react-scripts test --coverage --watchAll=false",
-    "storybook": "start-storybook --docs -p 6006 -s public",
+    "storybook": "cross-env NODE_OPTIONS=--openssl-legacy-provider start-storybook --docs -p 6006 -s public",
```
Comment (pconrad) [inline frontend/package.json:58]: Can you tell me more about this? What is it for and why is it needed?

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**